tests/components/modern_forms/test_config_flow.py

```diff
@@ -0,0 +1,198 @@
+"""Tests for the Modern Forms config flow."""
+from unittest.mock import MagicMock, patch
+
+import aiohttp
+from aiomodernforms import ModernFormsConnectionError
+
+from homeassistant.components.modern_forms.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER, SOURCE_ZEROCONF
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONTENT_TYPE_JSON
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import init_integration
+
+from tests.common import load_fixture
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_full_user_flow_implementation(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the full manual user flow from start to finish."""
+    aioclient_mock.post(
+        "http://192.168.1.123:80/mf",
+        text=load_fixture("modern_forms/device_info.json"),
+        headers={"Content-Type": CONTENT_TYPE_JSON},
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result.get("step_id") == "user"
+    assert result.get("type") == RESULT_TYPE_FORM
+    assert "flow_id" in result
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup functions on all create entry results to avoid setting up the whole integration.

Reply:
Example:
https://github.com/home-assistant/core/blob/30c53a1a13698c2592e91dd63e0305fa82d13f02/script/scaffold/templates/config_flow/tests/test_config_flow.py#L23-L26